#!/usr/bin/env python
import math
import os.path
from dynamic_reconfigure.parameter_generator_catkin import *

PI = round(math.pi, 4)

gen = ParameterGenerator()
gen.add("steering_shaft_x", double_t, 0, default=0.0,
        description="x coordinate (in m) of position of steering shaft in base frame")
gen.add("steering_shaft_y", double_t, 0, default=0.0,
        description="y coordinate (in m) of position of steering shaft in base frame")
gen.add("steering_crank_length", double_t, 0, default=0.0, min=0.0,
        description="length (in m) of crank connected to steering shaft")
gen.add("steering_crank_angle", double_t, 0, default=0.0, min=-PI, max=PI,
        description="angle (in rad) of crank connected to steering shaft, around z axis, measured from x axis")
gen.add("wheel_steering_arm_length", double_t, 0, default=0.0, min=0.0,
        description="length (in m) of arm connected to wheel kingpin")
gen.add("wheel_steering_arm_angle", double_t, 0, default=0.0, min=-PI, max=PI,
        description="angle (in rad) of arm connected to wheel kingpin, around z axis, measured from x axis")

package_name = os.path.basename(os.path.dirname(os.path.dirname(__file__)))
node_name = package_name + "_node"
cfg_name = os.path.splitext(os.path.basename(__file__))[0]
exit(gen.generate(package_name, node_name, cfg_name))
